### PR TITLE
Linter

### LINKED ISSUE #9: Implement linter

### PR DESCRIPTION
**Pull request by syky27**
Linter is now working Closes #9 

I saw some linting preferences written by someone somewhere, when I find them I will apply those on the config.

The build is failing, because MDs in repo are now not compliant with the linter config.

### CODE REVIEW

**Comment by syky27** [comment]
I have found @hroncok's reqs:

> Ok, I was thinking about using a Markdown linter, and this is what I would like to lint:
> - is the Markdown valid?
> - are there no HTML tags (Edux doesn't like them)?
> - are all link targets defined?
> - are all relative links and image sources existing?
> - are all nested lists always nested by exactly one level?

> Those are things I don't mind having linted:

> - line length, end of line marks, utf-8 encoding, no tabs
> - inline URLs

> And those are things I probably don't want to lint:
> 
> - what heading style is used (see #7 (comment))
> - whether * or - is used for lists, emphasis, etc.

**Comment by syky27** [comment]
@hroncok Would you be willing to check if the linting suites you? There are warning that occurs in your files, but nothing that would be in all of them. So those are inconsistencies.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 language: python
 python:
 - '3.6'
+node_js:
```
Comment (hroncok) [inline .travis.yml]: This seems to be ignored.

Comment (syky27) [reply .travis.yml]: Oh, yeah, you are correct, as @jiriutka mentioned in #5 node is already at the image.

Comment (syky27) [reply .travis.yml]: See: 17cec2e

**Comment by hroncok** [comment]
I'm going to rebase this.

**Comment by hroncok** [comment]
Rebased.

**Comment by hroncok** [comment]
Let me try to run the linter locally and fix the warnings / errors in our md files to see if I find the configuration useful.

**Review by syky27 — COMMENTED** [review]
Where did the fixup! came from, in the last commit message? Did you typed it in?

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "remark-validate-links": "^7.0.0"
     },
     "scripts": {
-        "fix": "remark cs/*.md -o",
-        "lint": "remark cs/*.md -f"
+        "fix": "remark *.md cs/*.md -o",
+        "lint": "remark *.md cs/*.md -f"
     }
```
Comment (syky27) [inline package.json]: Good point

**Comment by hroncok** [comment]
> Where did the fixup! came from, in the last commit message? Did you typed it in?

http://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html

**Comment by hroncok** [comment]
I've removed `remark-lint-blank-lines-1-0-2`, it insisted on 2 blank lines after all lists.

**Comment by syky27** [comment]
Ohhhh, fixup! is total GIT porn... 🎉

File: cs/course.md [hunk 3]
```diff
@@ -71,7 +71,7 @@ Semestrální práce
 Instalace programů
 ------------------
 
--   viz [Programy](apps.md)
+-   viz Programy (TODO dát link)
```
Comment (hroncok) [inline cs/course.md:74]: I'm not very happy with this, but let's just keep the linter happy for now, until we have it.

Comment (syky27) [reply cs/course.md:74]: I understand, but I believe that the leverage of having all links working just might overweight this...

You can always overcome this by creating empty target file, having only heading...

**Comment by hroncok** [comment]
OK, will rebase.

**Review by hroncok — APPROVED** [review]

**Comment by hroncok** [comment]
Thank you @mmajko, @syky27.

**Pull request merged**